src/javascript/app/Stores/Modules/Contract/contract-store.js

```diff
@@ -111,6 +111,8 @@ export default class ContractStore extends BaseStore {
         createChartBarrier(SmartChartStore, contract_info);
         createChartMarkers(SmartChartStore, contract_info);
 
+        SmartChartStore.updateMargin((end_time || contract_info.date_expiry) - date_start);
```

Comment:
should the margin always be sent together with the start and end epoch?

Reply:
it depends, even if we don't have start and end epoch yet we may have date_start and date_expiry so we can calculate the margin and send it